src/loops/nested_loops/hard/q8/MainTest.java

```diff
@@ -0,0 +1,57 @@
+package loops.nested_loops.hard.q8;
+
+import global.BaseTest;
+import global.exceptions.InvalidClauseException;
+import global.tools.TestOption;
+import global.variables.Clause;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.PlaceHolder;
+import global.variables.clauses.StringLiteral;
+import global.variables.wrappers.Optional;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+public class MainTest extends BaseTest {
+    // Java
+    public Clause[] testSentence(){
+        TestOption.isInputTest = true;
+        TestOption.defaultInput = "5";
+        return new Clause[]{
+                new StringLiteral("Enter a number:"),
+                new Optional(new StringLiteral(" ")),
+                new NewLine(),
+                new PlaceHolder()
+        };
+    }
+
+    public void runMain(){PrimeOrdial.main(new String[0]);}
+
+    static Stream<int[]> inputProvider(){
+        return Stream.of(new int[]{4, 10, 8, 100, 20, 7}, new int[]{6, 8, 1087}, new int[]{851, 4, 23}, new int[]{-5, -8, -7, -110, 3}, new int[]{0, 3}, new int[]{3});
```

Comment:
See my comment about negatives, and I think we should also add 1 to the one test because depending on how they code it, it could potentially pass and shouldn't
```suggestion
        return Stream.of(new int[]{4, 10, 8, 100, 20, 7}, new int[]{6, 8, 1087}, new int[]{851, 4, 23}, new int[]{-5, -8, -7, -110, 3}, new int[]{0, 1, 3}, new int[]{3});
```